process/slash/notifier/slashingNotifier.go

```diff
@@ -161,13 +173,35 @@ func (sn *slashingNotifier) computeTxData(proof coreSlash.SlashingProofHandler)
 	return []byte(dataStr), nil
 }
 
-func (sn *slashingNotifier) signTx(tx *transaction.Transaction) error {
+// Selects first KeyPair that is not in the same shard
+// where the slashing event has been detected
+func (sn *slashingNotifier) selectKeyPair() KeyPair {
+	var shardID uint32
+	selfID := sn.shardCoordinator.SelfId()
+
+	first := true
+	for currShardID := range sn.keyPairs {
```

Comment:
this selection looks a bit weird
I think you can remove the "first" variable and just initialize shardID with selfID

Reply:
>this selection looks a bit weird

Yes, I'm aware it looks weird, but only because I didn't want to iterate through the map anymore if I've already found a the first suitable keyPair. 
However, considering that the map containing the `keyPairs` will most likely be very small in size (most probably 3 wallets, one in each shard) I can change the code to make more readable.

>I think you can remove the "first" variable and just initialize shardID with currShardID

Right, applied your suggestion